docs/how-to/extending/sql.qmd

```diff
@@ -0,0 +1,186 @@
+```
+
+1. Connect to an in-memory DuckDB database
+2. Read in the `penguins` example with our DuckDB database, and name it `penguins`
+
+## `Table.sql`
+
+At the highest level there's the `Table.sql` method. This method allows you to
+run arbitrary `SELECT` statements against a table expression:
+
+```{python}
+t.sql("SELECT * FROM penguins")
+```
+
+Now, `SELECT *` on a table expressions is not particularly useful and in fact
+is a bit wasteful: there's no need to reselect a table expression.
+
+The real power of `Table.sql` is composing it with other ibis expressions:
+
+```{python}
+(
+   t.sql(
+      """
+      SELECT
+        species,
+        island,
+        mad(bill_length_mm) AS bill_mad
+      FROM penguins
+      GROUP BY 1, 2
+      """
+   )
+   .filter(_.bill_mad > 2)
+   .order_by(_.bill_mad.desc())
+)
+```
+
+This method is powerful and you can mix and match Ibis expressions as you like:
+
+```{python}
+(
+   t.sql(
+      """
+      SELECT
+        species,
+        island,
+        mad(bill_length_mm) AS bill_mad
+      FROM penguins
+      GROUP BY 1, 2
+      """
+   )
+   .filter(_.bill_mad > 2)
+   .alias("big_bills")  # note the alias call for subsequent '.sql' calls
+   .sql("SELECT * FROM big_bills ORDER BY 3 DESC")
+)
+```
+
+### Working with different SQL dialects
+
+You can also pass SQL strings from SQL dialects that do not match the backend
+you're using by passing a dialect name to the `dialect` argument of `.sql`.
+
+For example, here's MySQL syntax running against DuckDB (note the use of
+backticks for quoting).
+
+```{python}
+(
+   t.sql(
+      """
+      SELECT
+        `species`,
+        `island`,
+        mad(`bill_length_mm`) AS bill_mad
+      FROM `penguins`
+      GROUP BY 1, 2
+      """,
+      dialect="mysql",
+   )
+   .filter(_.bill_mad > 2)
+   .alias("big_bills")  # note the alias call for subsequent '.sql' calls
+   .sql("SELECT * FROM big_bills ORDER BY 3 DESC")  # <1>
+)
+```
+
+1. By default the dialect is the backend's native dialect.
+
+This feature is useful if you're porting existing SQL from one backend to
+another.
+
+## `Backend.sql`
+
+There's also the `Backend.sql` method, which can handle arbitrary `SELECT`
+statements as well and return's an Ibis table expression.
+
+The main difference with `Table.sql` is that `Backend.sql` **can only refer to
+tables that already exist in the database**, because the API is defined on
+`Backend` instances.
+
+After the `Backend.sql` call, however, you're able to mix and match similar
+to `Table.sql`:
+
+```{python}
+(
+   con.sql(
+      """
+      SELECT
+        species,
+        island,
+        mad(bill_length_mm) AS bill_mad
+      FROM penguins
+      GROUP BY 1, 2
+      """
+   )
+   .filter(_.bill_mad > 2)
+   .alias("big_bills")  # note the alias call for subsequent '.sql' calls
+   .sql("SELECT * FROM big_bills ORDER BY 3 DESC")
+)
+```
+
+::: {.callout-tip}
+## `Backend.sql` also supports the `dialect` argument.
```

Comment:
This callout doesn't feel actionable for the user - what is the `dialect` argument and why might they want to use it? A brief description or example here would be helpful IMO.

Reply:
I discuss it above in `Table.sql`